src/esp/gfx/RenderCamera.cpp

```diff
@@ -77,18 +77,22 @@ uint32_t RenderCamera::draw(MagnumDrawableGroup& drawables,
     return drawables.size();
   }
 
-  std::vector<std::pair<std::reference_wrapper<Mn::SceneGraph::Drawable3D>,
-                        Mn::Matrix4>>
-      drawableTransforms = drawableTransformations(drawables);
+  // dt is short for drawable transformations
+  using dt =
+      std::vector<std::pair<std::reference_wrapper<Mn::SceneGraph::Drawable3D>,
+                            Mn::Matrix4>>;
+  dt drawableTransforms = drawableTransformations(drawables);
 
   // draw just the visible part
   size_t numVisibles = cull(drawableTransforms);
-  // erase all items that did not pass the frustum visibility test
-  drawableTransforms.erase(drawableTransforms.begin() + numVisibles,
-                           drawableTransforms.end());
 
-  MagnumCamera::draw(drawableTransforms);
-  return drawableTransforms.size();
+  dt::iterator iter = drawableTransforms.begin();
+  for (size_t iDrawable = 0; iDrawable < numVisibles; ++iDrawable) {
+    iter->first.get().draw(iter->second, *this);
+    iter++;
+  }
```

Comment:
Does *this* actually improve anything? I doubt that, since `std::pair<std::reference_wrapper<Mn::SceneGraph::Drawable3D>, Mn::Matrix4>` has a trivial destructor and with optimizations enabled, `erase()` collapses to just adjusting vector size.

Reply:
This is just the consequence if I would like to get rid of the "erase()". 
Because if I do not use "erase()", the end() will still point to the old end position, and you will draw all the "invisible ones" at the 2nd half of the vector.

Reply:
You could use `erase()` even with a preallocated vector -- it won't free any memory, `capacity()` will stay the same after.

Reply:
If I use `erase()` it will call the destructor of the to-be-deleted items, which I tried to save.
It has nothing to do with the "deallocator".

Reply:
But like I said, it will not help you much since this is a local vector, and all the elements within will be destroyed (the destructor will be called) sooner or later.

Reply:
I am not sure if grouping up the destructor calls really matters.  The destructor will always need to run, even if it is a global vector.

Reply:
@erikwijmans : Aha, so even if this is a global vector, when we initialize it every frame, the "old value" from last frame will get destroyed no matter what. I see.

Reply:
> it will call the destructor of the to-be-deleted items

@bigbike Just to reiterate what I said earlier and avoid confusion/misunderstanding -- the `std::pair<std::reference_wrapper<Mn::SceneGraph::Drawable3D>, Mn::Matrix4>` is a trivially-destructible type, which means there's no code at all to run at destruction, so the destructor calls will get optimized out.